src/Mapster/Adapters/ClassAdapter.cs

```diff
@@ -23,10 +23,11 @@ protected override bool CanMap(Type sourceType, Type destinationType, MapType ma
             if (sourceType == typeof(string) || sourceType == typeof(object))
                 return false;
 
-            if (!destinationType.IsPoco())
+            if (destinationType.GetTypeInfo().IsEnum || destinationType.Namespace.StartsWith("System"))
                 return false;
 
-            return true;
+            return (destinationType.GetPublicFieldsAndProperties(allowNoSetter: false).Any() ||
+                    destinationType.GetPrivateFieldsAndProperties(allowNoSetter: false).Any());
```

Comment:
Yes, score is precedence, higher score will apply when it pass `CanMap` method. I think score for `ClassWithNonPublicMemberAdapter` could be any, since there is no adapter that can map private fields.